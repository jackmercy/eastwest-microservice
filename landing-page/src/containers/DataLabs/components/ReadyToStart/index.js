import dynamic from 'next/dynamic';

export default dynamic(() => import('./ReadyToStart'), { ssr: false });
