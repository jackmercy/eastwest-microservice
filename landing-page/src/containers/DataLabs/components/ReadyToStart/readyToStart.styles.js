import { createUseStyles } from 'react-jss';

import COLORS from '../../../../constants/COLORS';
import { hexToRgbA, rem } from '../../../../utils/jss';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: { backgroundColor: hexToRgbA(COLORS.bgBlack01, 0.02) },
  contentWrapper: { paddingTop: rem(24), paddingBottom: rem(120) },
  content: { position: 'relative' },
  image: { display: 'block', width: '100%' },
  group: {
    position: 'absolute',
    left: rem(64),
    top: rem(56),
    maxWidth: rem(400),
  },
  title: {
    fontWeight: 900,
    fontSize: rem(48),
    lineHeight: rem(64),
    margin: 0,
  },
  button: { marginTop: rem(40) },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxMd}px)`]: {
    root: { backgroundColor: COLORS.bgPink },
    contentWrapper: { paddingTop: rem(20), paddingBottom: rem(32) },
    group: { position: 'static', maxWidth: '100%', textAlign: 'center' },
    title: { marginTop: rem(24) },
    image: {
      maxWidth: rem(375),
      marginLeft: 'auto',
      marginRight: 'auto',
      '& + $image': { marginTop: rem(16) },
    },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    title: { fontSize: rem(32), lineHeight: rem(40) },
    button: { marginTop: rem(30) },
  },
});
