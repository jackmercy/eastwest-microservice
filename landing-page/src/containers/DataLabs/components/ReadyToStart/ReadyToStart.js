import Link from 'next/link';
import { memo } from 'react';
import { useIntl } from 'react-intl';
import PropTypes from 'prop-types';
import { useMediaQuery } from 'react-responsive';

import { Content } from '../../../components';
import { Button } from '../../../../components';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

import useStyles from './readyToStart.styles';

const ReadyToStart = ({ defaultReason }) => {
  const intl = useIntl();
  const classes = useStyles();
  const isMobile = useMediaQuery({
    query: `(max-width: ${BREAKPOINTS.maxMd}px)`,
  });

  return (
    <div className={classes.root}>
      <Content className={classes.contentWrapper}>
        <div className={classes.content}>
          {!isMobile ? (
            <img
              alt={intl.formatMessage({ id: 'DATA_LABS.READY_TO_START.TITLE' })}
              src="/static/images/data-labs/get-start-01.png"
              className={classes.image}
            />
          ) : (
            <>
              <img
                alt={intl.formatMessage({
                  id: 'DATA_LABS.READY_TO_START.TITLE',
                })}
                src="/static/images/data-labs/get-start-mobile-01.png"
                className={classes.image}
              />
              <img
                alt={intl.formatMessage({
                  id: 'DATA_LABS.READY_TO_START.TITLE',
                })}
                src="/static/images/data-labs/get-start-mobile-02.png"
                className={classes.image}
              />
            </>
          )}
          <div className={classes.group}>
            <h2 className={classes.title} data-test="ready-to-start-title">
              {intl.formatMessage({ id: 'DATA_LABS.READY_TO_START.TITLE' })}
            </h2>
            <Link
              href={`/contact/?defaultReason=${defaultReason}#contact-form`}
              passHref
              as="/contact/#contact-form"
            >
              <Button
                tag="a"
                color="primary"
                size="large"
                keepSizeOnMobile
                className={classes.button}
                dataTest="ready-to-start-button"
              >
                {intl.formatMessage({ id: 'DATA_LABS.READY_TO_START.BUTTON' })}
              </Button>
            </Link>
          </div>
        </div>
      </Content>
    </div>
  );
};
ReadyToStart.propTypes = {
  defaultReason: PropTypes.number.isRequired,
};

export default memo(ReadyToStart);
