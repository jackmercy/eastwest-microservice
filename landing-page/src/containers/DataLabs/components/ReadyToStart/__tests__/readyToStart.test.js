import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ReadyToStart from '../ReadyToStart';

describe('Testing DataLabs components > ReadyToStart', () => {
  let wrapper;
  const props = { defaultReason: 4 };

  beforeAll(() => {
    wrapper = mount(<ReadyToStart {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<ReadyToStart {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="ready-to-start-title"]')).toHaveLength(1);
  });

  it('should have a button', () => {
    expect(wrapper.find('[data-test="ready-to-start-button"]')).toHaveLength(1);
  });
});
