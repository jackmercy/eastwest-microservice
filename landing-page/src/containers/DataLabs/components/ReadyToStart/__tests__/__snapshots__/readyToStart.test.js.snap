// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DataLabs components > ReadyToStart should render match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <Memo(Content)
    className="contentWrapper-0-2-2"
  >
    <div
      className="content-0-2-3"
    >
      <img
        alt="Ready to start use Data Labs?"
        className="image-0-2-4"
        src="/static/images/data-labs/get-start-01.png"
      />
      <div
        className="group-0-2-5"
      >
        <h2
          className="title-0-2-6"
          data-test="ready-to-start-title"
        >
          Ready to start use Data Labs?
        </h2>
        <Link
          as="/contact/#contact-form"
          href="/contact/?defaultReason=4#contact-form"
          passHref={true}
        >
          <Memo([object Object])
            className="button-0-2-7"
            color="primary"
            dataTest="ready-to-start-button"
            keepSizeOnMobile={true}
            size="large"
            tag="a"
          >
            Get started
          </Memo([object Object])>
        </Link>
      </div>
    </div>
  </Memo(Content)>
</div>
`;
